Sources/Commands/SwiftPackageTool.swift

```diff
@@ -999,6 +988,9 @@ extension SwiftPackageTool {
                 writableDirectories.append(AbsolutePath(pathString, relativeTo: swiftTool.originalWorkingDirectory))
             }
 
+            // Make sure that the package path is read-only unless it's covered by any of the explicitly writable directories.
+            let readOnlyDirectories = writableDirectories.contains{ package.path.isDescendantOfOrEqual(to: $0) } ? [] : [package.path]
+
```

Comment:
This one is subject to debate.  It avoids marking the package path as readonly if it's covered by any of the directories marked as explicitly writable.  Maybe what should be done here in the long term is to have a single list of directory-specific permissions, each of which could be either `block` or `allow` and then the caller can craft their sandbox directories as they want to.  For example, it might make sense to allow a plugin to modify only a subdirectory of the package.  We can extend this in the future — probably better to keep it simple for now.

Reply:
starting simple makes sense to me

Reply:
👍